### PR TITLE
JDK-8292576: Improve wording of AccessFlag-related specs

### PR DESCRIPTION
**Pull request by jddarcy**
Spec refinement, including acting on some feedback received for JDK-8266670:

https://github.com/openjdk/jdk/pull/7445#discussion_r906846794

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292576](https://bugs.openjdk.org/browse/JDK-8292576): Improve wording of AccessFlag-related specs

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9912/head:pull/9912` \
`$ git checkout pull/9912`

Update a local copy of the PR: \
`$ git checkout pull/9912` \
`$ git pull https://git.openjdk.org/jdk pull/9912/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9912`

View PR using the GUI difftool: \
`$ git pr show -t 9912`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9912.diff">https://git.openjdk.org/jdk/pull/9912.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/module/ModuleDescriptor.java [hunk 1]
```diff
@@ -214,7 +214,7 @@ public Set<Modifier> modifiers() {
 
         /**
          * {@return an unmodifiable set of the module {@linkplain AccessFlag
-         * requires flags, possibly empty}}
+         * requires flags}, possibly empty}
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/lang/module/ModuleDescriptor.java:217]: The link fix looks okay but the wording in the new accessFlags() method is different to the wording in the other methods. The other methods use "possibly-empty unmodifiable" in the return description. Either is fine but I think we should try to keep the wording consistent where possible.

Comment (jddarcy) [reply src/java.base/share/classes/java/lang/module/ModuleDescriptor.java:217]: I've looked over the wording of the various accessFlags methods. The wording for ModuleDescriptor.Requires looks consistent to me:

Returns an unmodifiable set of the module flags, possibly empty.

Returns an unmodifiable set of the module export flags for this module descriptor, possibly empty.

Returns an unmodifiable set of the module opens flags, possibly empty.

Returns an unmodifiable set of the module requires flags, possibly empty.

----

Returns an unmodifiable set of the access flags for this class, possibly empty.

Returns an unmodifiable set of the access flags for the executable represented by this object, possibly empty.

Returns an unmodifiable set of the access flags for this member, possibly empty.

Returns an unmodifiable set of the access flags for the parameter represented by this object, possibly empty.

Am I overlooking something in the twisty passages, all alike?

Comment (AlanBateman) [reply src/java.base/share/classes/java/lang/module/ModuleDescriptor.java:217]: If you compare the javadoc for the modifiers vs. accessFlags in each of ModuleDescriptor, Requires, Exports, and Opens then you'll see what I mean. I was hoping they could be inconsistent.

Comment (jddarcy) [reply src/java.base/share/classes/java/lang/module/ModuleDescriptor.java]: > If you compare the javadoc for the modifiers vs. accessFlags in each of ModuleDescriptor, Requires, Exports, and Opens then you'll see what I mean. I was hoping they could be consistent.

Ah, okay. Pushed docs updated as suggested for consistency within ModuleDescriptor.java.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**